src/pages/index.js

```diff
@@ -1,24 +1,103 @@
+import React, { useContext, useEffect, useState } from "react"
+import useInterval from "../hooks/useInterval"
+
+import { Link } from "gatsby"
+import { HeartwoodStateContext } from "../state/HeartwoodContextProvider"
+import { isLoggedIn, signUpComplete } from "../utils/utils"
+import gear from "../assets/gear.svg"
+
 import Navbar from "../components/navbar"
-import { navigate } from "gatsby"
-import { HeartwoodStateContext, HeartwoodDispatchContext } from "../state/HeartwoodContextProvider"
-import { isLoggedIn } from "../utils/utils"
+import CollectInfo from "../components/collectinfo"
+
+import { format, differenceInMinutes } from "date-fns"
 
 const IndexPage = () => {
   const state = useContext(HeartwoodStateContext)
-  const dispatch = useContext(HeartwoodDispatchContext)
+  const [form, setForm] = useState({ startTime: "", stopTime: "", showMeetingButton: false, lessonDescription: "", isLoading: true})
 
-  useEffect(() => {
-    if (isLoggedIn(state.user)) {
-      const { photoURL, displayName, email } = state.user
+  const setMeetingButton = () => {
+    const currentTime = new Date()
+    // Check if the current time is within five minutes either way of the next lesson's time
+    if (differenceInMinutes(form.startTime, currentTime) <= 5 && differenceInMinutes(currentTime, form.stopTime) <= 5) {
+      setForm({ ...form, showMeetingButton: true, isLoading: false })
     } else {
-      navigate("/signin")
+      setForm({ ...form, showMeetingButton: false, isLoading: false })
+    }
+  }
+  
+  useEffect(() => {
+    if (state.user) {
+      console.log('in useEffect', form)
+      fetch("http://localhost:5000/getNextMeeting", {method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ uid: state.user.uid })})
+        .then((res) => res.json())
+        .then((res) => {
+          console.log("response has arrived", res)
+          if (res.startTime === -1 && res.stopTime === -1) {
+            // catch case where there is not a next meeting
+            setForm({...form, 
+              lessonDescription: "You do not have any upcoming lessons scheduled"})
+          } else {
+            console.log("its a meeting")
+            // otherwise handle the case where a next meeting exists
+            const formattedStartTime = new Date(res.startTime.dateTime)
+            const formattedStopTime = new Date(res.stopTime.dateTime)
+            const lessonDescription = format(formattedStartTime,
+              "'Your next lesson is on: ' EEEE, LLLL do, y 'at' h:mm a '(PST)'")
+            setForm({...form, startTime: formattedStartTime, stopTime: formattedStopTime, lessonDescription: lessonDescription})
+            }
+        })
+        .catch((res) => {
+          console.error("unable to get next meeting")
+        })
     }
   }, [state.user])
 
+  useEffect(() => {
+    if (form.lessonDescription !== 'DEFAULT') {
+      setMeetingButton()
+    }
+  }, [form.lessonDescription])
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'setMeetingButton'. Either include it or remove the dependency array.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5837031)